applications/GeoMechanicsApplication/custom_constitutive/incremental_linear_elastic_interface_law.cpp

```diff
@@ -0,0 +1,131 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "incremental_linear_elastic_interface_law.h"
+#include "custom_geometries/line_interface_geometry.h"
+#include "geo_mechanics_application_constants.h"
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+ConstitutiveLaw::Pointer GeoIncrementalLinearElasticInterfaceLaw::Clone() const
+{
+    return std::make_shared<GeoIncrementalLinearElasticInterfaceLaw>(*this);
+}
+
+ConstitutiveLaw::SizeType GeoIncrementalLinearElasticInterfaceLaw::WorkingSpaceDimension()
+{
+    return 2;
+}
+
+ConstitutiveLaw::SizeType GeoIncrementalLinearElasticInterfaceLaw::GetStrainSize() const
+{
+    return VOIGT_SIZE_2D_INTERFACE;
+}
+
+Vector& GeoIncrementalLinearElasticInterfaceLaw::GetValue(const Variable<Vector>& rThisVariable, Vector& rValue)
+{
+    if (rThisVariable == STRAIN) {
+        rValue = mPreviousRelativeDisplacement;
+    } else if (rThisVariable == CAUCHY_STRESS_VECTOR) {
+        rValue = mPreviousTraction;
+    }
+
+    return rValue;
+}
+
+ConstitutiveLaw::StressMeasure GeoIncrementalLinearElasticInterfaceLaw::GetStressMeasure()
+{
+    return ConstitutiveLaw::StressMeasure_Cauchy;
+}
+
+bool GeoIncrementalLinearElasticInterfaceLaw::IsIncremental() { return true; }
+
+void GeoIncrementalLinearElasticInterfaceLaw::InitializeMaterial(const Properties&,
+                                                                 const ConstitutiveLaw::GeometryType&,
+                                                                 const Vector&)
+{
+    mPreviousRelativeDisplacement =
+        HasInitialState() ? GetInitialState().GetInitialStrainVector() : ZeroVector{GetStrainSize()};
+    mPreviousTraction =
+        HasInitialState() ? GetInitialState().GetInitialStressVector() : ZeroVector{GetStrainSize()};
+}
+
+void GeoIncrementalLinearElasticInterfaceLaw::CalculateMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    rValues.GetStressVector() =
+        mPreviousTraction +
+        prod(MakeConstitutiveMatrix(rValues.GetMaterialProperties()[INTERFACE_NORMAL_STIFFNESS],
+                                    rValues.GetMaterialProperties()[INTERFACE_SHEAR_STIFFNESS]),
+             rValues.GetStrainVector() - mPreviousRelativeDisplacement);
```

Comment:
Very clear representation of the formula

Reply:
Thank you. That was our intention :-)